### ISSUE TITLE
training problem with rnn_transducer

### ISSUE THREAD

**Issue by yiqiaoc11** [TensorSpeech/TensorFlowASR#279]
The transducer in TensorFlowASR\examples\rnn_transducer doesn't work for the current version with either current or pretrained config.yml. This is a fundamental function. Can the author or someone give it a try to validate it?

**Comment by usimarit** [comment]
Hi @yiqiaoc11 
I'm training on TPUs to validate this.
Are you using the warp-transducer loss or the rnnt loss in tensorflow?
So far as I'm testing with rnnt loss in tensorflow for the past months, it has some issues with convergence. But I dont have resources to test with GPUs.

**Comment by usimarit** [comment]
@yiqiaoc11 Could you help me train 2 models for 30 epochs using rnnt-loss:

1. 4-layers encoder
2. 8-layers encoder

Then plot the loss of 2 models for better comparison?
Other config are the same.

**Comment by yiqiaoc11** [comment]
@usimarit,  using the config.yml (https://github.com/TensorSpeech/TensorFlowASR/blob/main/examples/rnn_transducer/config.yml) without warmup_steps, right?

**Comment by usimarit** [comment]
@yiqiaoc11 Yes, with the pretrained config

I trained the rnn transducer on TPUs so warp-transducer loss cannot be applied, only rnnt-loss can be used here. But you can experiment with warp-transducer loss too, plotting the loss of 2 models for better comparison.

**Comment by usimarit** [comment]
@yiqiaoc11 The rnn_transducer structure stays the same in version v1.0.x
Is the number of weights in your case the same as in the pretrained example?

**Comment by yiqiaoc11** [comment]
Yes, the number of weights and distributions of layers are same, but other config information from the pretrained isn't tractable. Not sure what leads to the underfitting observed.

Primary loss curves for 4/8layer are posted for differentiation. Green curves are for 8-layer while blue 4-layer. Losses are very similar while models were tuned under the same .yml in GDrive. They don't converge.
![Untitled](https://user-images.githubusercontent.com/30539007/218009198-445475f4-4ebe-49e2-957d-6f1853166f91.png)



[2023-02-09 09:03:10] PRINT  Layer (type)                                Output Shape                            Param #        
[2023-02-09 09:03:10] PRINT ====================================================================================================
[2023-02-09 09:03:10] PRINT  streaming_transducer_encoder_reshape (Resha  multiple                               0              
[2023-02-09 09:03:10] PRINT  pe)                                                                                                
[2023-02-09 09:03:10] PRINT                                                                                                     
[2023-02-09 09:03:10] PRINT  streaming_transducer_encoder_block_0 (RnnTr  multiple                               5511488        
[2023-02-09 09:03:10] PRINT  ansducerBlock)                                                                                     
[2023-02-09 09:03:10] PRINT                                                                                                     
[2023-02-09 09:03:10] PRINT  streaming_transducer_encoder_block_1 (RnnTr  multiple                               7149888        
[2023-02-09 09:03:10] PRINT  ansducerBlock)                                                                                     
[2023-02-09 09:03:10] PRINT                                                                                                     
[2023-02-09 09:03:10] PRINT  streaming_transducer_encoder_block_2 (RnnTr  multiple                               5839168        
[2023-02-09 09:03:10] PRINT  ansducerBlock)                                                                                     
[2023-02-09 09:03:10] PRINT                                                                                                     
[2023-02-09 09:03:10] PRINT  streaming_transducer_encoder_block_3 (RnnTr  multiple                               5839168        
[2023-02-09 09:03:10] PRINT  ansducerBlock)                                                                                     
[2023-02-09 09:03:10] PRINT                                                                                                     
[2023-02-09 09:03:10] PRINT ====================================================================================================
[2023-02-09 09:03:10] PRINT Total params: 24,339,712
[2023-02-09 09:03:10] PRINT Trainable params: 24,339,712
[2023-02-09 09:03:10] PRINT Non-trainable params: 0


[2023-02-09 09:03:15] PRINT  Layer (type)                                Output Shape                            Param #        
[2023-02-09 09:03:15] PRINT ====================================================================================================
[2023-02-09 09:03:15] PRINT  streaming_transducer_encoder_reshape (Resha  multiple                               0              
[2023-02-09 09:03:15] PRINT  pe)                                                                                                
[2023-02-09 09:03:15] PRINT                                                                                                     
[2023-02-09 09:03:15] PRINT  streaming_transducer_encoder_block_0 (RnnTr  multiple                               5511488        
[2023-02-09 09:03:15] PRINT  ansducerBlock)                                                                                     
[2023-02-09 09:03:15] PRINT                                                                                                     
[2023-02-09 09:03:15] PRINT  streaming_transducer_encoder_block_1 (RnnTr  multiple                               7149888        
[2023-02-09 09:03:15] PRINT  ansducerBlock)                                                                                     
[2023-02-09 09:03:15] PRINT                                                                                                     
[2023-02-09 09:03:15] PRINT  streaming_transducer_encoder_block_2 (RnnTr  multiple                               5839168        
[2023-02-09 09:03:15] PRINT  ansducerBlock)                                                                                     
[2023-02-09 09:03:15] PRINT                                                                                                     
[2023-02-09 09:03:15] PRINT  streaming_transducer_encoder_block_3 (RnnTr  multiple                               5839168        
[2023-02-09 09:03:15] PRINT  ansducerBlock)                                                                                     
[2023-02-09 09:03:15] PRINT                                                                                                     
[2023-02-09 09:03:15] PRINT  streaming_transducer_encoder_block_4 (RnnTr  multiple                               5839168        
[2023-02-09 09:03:15] PRINT  ansducerBlock)                                                                                     
[2023-02-09 09:03:15] PRINT                                                                                                     
[2023-02-09 09:03:15] PRINT  streaming_transducer_encoder_block_5 (RnnTr  multiple                               5839168        
[2023-02-09 09:03:15] PRINT  ansducerBlock)                                                                                     
[2023-02-09 09:03:15] PRINT                                                                                                     
[2023-02-09 09:03:15] PRINT  streaming_transducer_encoder_block_6 (RnnTr  multiple                               5839168        
[2023-02-09 09:03:15] PRINT  ansducerBlock)                                                                                     
[2023-02-09 09:03:15] PRINT                                                                                                     
[2023-02-09 09:03:15] PRINT  streaming_transducer_encoder_block_7 (RnnTr  multiple                               5839168        
[2023-02-09 09:03:15] PRINT  ansducerBlock)                                                                                     
[2023-02-09 09:03:15] PRINT                                                                                                     
[2023-02-09 09:03:15] PRINT ====================================================================================================
[2023-02-09 09:03:15] PRINT Total params: 47,696,384
[2023-02-09 09:03:15] PRINT Trainable params: 47,696,384
[2023-02-09 09:03:15] PRINT Non-trainable params: 0
